### PR TITLE
feat(native): add a compressed minidump endpoint example

### LINKED ISSUE #7575: Add compressed minidump example.
### Core or SDK?

Core Sentry product

### Which part? Which one?

https://docs.sentry.io/platforms/native/guides/minidumps/#minidump-integration

### Description

Some users might run into 413 request payload too large when sending minidumps using the minidump endpoint. One solution is to compress the file as gzip to work around the size limitation, but there is no example request.

### Suggested Solution

Add the following example:
```
curl -X POST   'minidump_endpoint'   --data-binary @mini.dmp.gz \ 
-H 'content-type:application/x-dmp' \
-H 'content-encoding: gzip'
```

### PR DESCRIPTION
**Pull request by supervacuus**
## DESCRIBE YOUR PR

The docs currently do not describe how the minidump endpoint can be used when sending compressed payloads. This is often necessary when customers produce minidumps bigger than our upload limits.

Fixes #7575.

## IS YOUR CHANGE URGENT?  

Help us prioritize incoming PRs by letting us know when the change needs to go live.
- [ ] Urgent deadline (GA date, etc.): 
- [ ] Other deadline: 
- [x] None: Not urgent, can wait up to 1 week+

## SLA

- Teamwork makes the dream work, so please add a reviewer to your PRs.
- Please give the docs team up to 1 week to review your PR unless you've added an urgent due date to it.
Thanks in advance for your help!

## PRE-MERGE CHECKLIST

*Make sure you've checked the following before merging your changes:*

- [x] Checked Vercel preview for correctness, including links
- [ ] PR was reviewed and approved by any necessary SMEs (subject matter experts)
- [x] PR was reviewed and approved by a member of the [Sentry docs team](https://github.com/orgs/getsentry/teams/docs)

### CODE REVIEW

**Comment by supervacuus** [comment]
I added @markushi and @JoshuaMoelans since the basic compressed example is relatively simple, but adding meta-data is quite involved.

I would love to know if the steps and reasoning are understandable, provided the user is technically sophisticated enough to use `curl` with a multipart endpoint.

File: docs/platforms/native/guides/minidumps/index.mdx
Comment (vivianyentran) [inline docs/platforms/native/guides/minidumps/index.mdx:116]:
```suggestion
You can compress your minidump using `gzip` if its size exceeds the [limits](#size-limits) in the next section.
```

File: docs/platforms/native/guides/minidumps/index.mdx
Comment (vivianyentran) [inline docs/platforms/native/guides/minidumps/index.mdx:132]:
```suggestion
Since `curl` doesn't allow adding multipart form fields to "data" `POST` requests, you can't easily pass additional data as fields as in the previous section's examples.

To get around this, you must construct the multipart content manually and compress it as a whole:
```

**Review by vivianyentran — APPROVED** [review]

File: docs/platforms/native/guides/minidumps/index.mdx
Comment (JoshuaMoelans) [inline docs/platforms/native/guides/minidumps/index.mdx:141]: My first time going through these steps I didn't change this "release" to the actual version that the .dmp file was built for, which I think is the reason I got an `{"detail":"invalid minidump"}%` result. 

Not sure if the advanced users that would use this method of sending minidumps would make the same mistake, but still might be something worth highlighting.

Other than that, these instructions seem fine as a beginner's guide.

**Review by JoshuaMoelans — COMMENTED** [review]
Steps are reasonable (even for someone who's new to this). 

**Error below was resolved**
 (turns out I used ” instead of ")

---

I am wondering if this warning on the dashboard below is supposed to show up?
Screenshot 2024-09-11 at 14 34 54

**Comment by supervacuus** [comment]
Following the conversation here: https://github.com/getsentry/sentry-docs/pull/11304#discussion_r1754783544 I created a Relay-PR (https://github.com/getsentry/relay/pull/4029) that would reduce this documentation change to:

> If you want to compress your minidump, you can `gzip` it and upload the `*.gz` instead. All previous examples apply.

Let's wait before merging if the Relay folk consider this a sensible idea.

**Comment by supervacuus** [comment]
I minimized the docs to the outcome of https://github.com/getsentry/relay/pull/4029, which the relay team merged today. Let's wait until it is deployed so we can test it before we merge the changes to the docs.

**Comment by supervacuus** [comment]
I tested this with all four supported compression containers against a public ingest endpoint, and it worked as expected. 

If you do not have any further adaptations (given the drastic content reduction compared to the initial proposal), @vivianyentran, and if the short version also makes sense to you, @JoshuaMoelans, I feel this is good to merge.

**Comment by JoshuaMoelans** [comment]
lgtm!

**Review by JoshuaMoelans — APPROVED** [review]

**Pull request merged**
